src/route-sync/cloudfoundry/tcp/router_test.go

```diff
@@ -0,0 +1,156 @@
+package tcp_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	. "route-sync/cloudfoundry/tcp"
+	"route-sync/route"
+
+	"code.cloudfoundry.org/uaa-go-client"
+	uaafakes "code.cloudfoundry.org/uaa-go-client/fakes"
+	uaaschema "code.cloudfoundry.org/uaa-go-client/schema"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("routing-api TCP router", func() {
+	var (
+		fooClient uaafakes.FakeClient
+	)
+	BeforeEach(func() {
+		fooClient = uaafakes.FakeClient{}
+		fooClient.FetchTokenReturns(&uaaschema.Token{
+			AccessToken: "foo",
+			ExpiresIn:   0,
+		}, nil)
+	})
+
+	It("requires UAA client and an API endpoint", func() {
+		var invalidCreations = []struct {
+			uaaClient uaa_go_client.Client
+			api       string
+		}{
+			{nil, ""},
+			{&fooClient, ""},
+			{nil, "bar"},
+		}
+
+		for _, t := range invalidCreations {
+			Context(fmt.Sprintf("uaaClient: %q, api: %s", t.uaaClient, t.api), func() {
+				router, err := NewRoutingApi(t.uaaClient, t.api, false)
+				Expect(router).To(BeNil())
+				Expect(err).NotTo(BeNil())
+			})
+		}
+
+		router, err := NewRoutingApi(&fooClient, "foo", false)
```

Comment:
That's true, in fact I really don't see the value of this positive test. It just performs creation.